src/kibana/directives/saved_object_finder.js

```diff
@@ -104,100 +108,121 @@ define(function (require) {
+              self.selector.index = -1;
+              self.selector.enabled = false;
 
-              // check if there is a url for this hit
-              var url = $scope.makeUrl(hit);
+              if ($event.shiftKey) {
+                $input.focus();
+              }
 
-              if (url) window.location = url;
-              $scope.onChoose(hit);
+              $event.preventDefault();
+              break;
+            case 'down':
+              if (!self.selector.enabled) break;
 
-              return;
-            case up:
-              $next = $selected ? $selected.prev() : $list.find('li:last-child');
-              goingUp = false;
+              if (self.selector.index + 1 < page.length) {
+                self.selector.index += 1;
+              }
+              $event.preventDefault();
               break;
-            case down:
-              $next = $selected ? $selected.next() : $list.find('li:first-child');
-              goingUp = true;
+            case 'up':
+              if (!self.selector.enabled) break;
+
+              if (self.selector.index > 0) {
+                self.selector.index -= 1;
+              }
+              $event.preventDefault();
               break;
-            case esc:
-              scrollIntoView($list.find('li:first-child'));
-              $next = null;
+            case 'right':
+              if (!self.selector.enabled) break;
+
+              if (page.number < page.count) {
+                paginate.goToPage(page.number + 1);
+                self.selector.index = 0;
+                $timeout(function () {
+                  $list.find('a:first').focus();
+                });
+              }
+              $event.preventDefault();
               break;
-            default:
-              return;
-            }
-
-            if ($next && $next.length === 0) {
-              // we are at one of the ends
-              return;
-            }
+            case 'left':
+              if (!self.selector.enabled) break;
+
+              if (page.number > 1) {
+                paginate.goToPage(page.number - 1);
+                self.selector.index = 0;
+                $timeout(function () {
+                  $list.find('a:first').focus();
+                });
+              }
+              $event.preventDefault();
+              break;
+            case 'escape':
+              if (!self.selector.enabled) break;
 
-            if ($selected && $next && $next.eq($selected).length) {
-              // the selections are the same, bail
-              return;
-            }
+              $input.focus();
+              $event.preventDefault();
+              break;
+            case 'enter':
+              if (!self.selector.enabled) break;
 
-            if ($selected) {
-              $selected.removeClass('active');
-              $selected = null;
-            }
+              var hitIndex = ((page.number - 1) * paginate.perPage) + self.selector.index;
```

Comment:
The function that is the meat of the work being done is called both by this handler and bound to the click event of each hit in the markup. Per our discussion, you agreed that this was fine.